src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,52 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MAX_PASSWORD_LENGTH = 20;
+    private static final String EMPTY_SPACE_REGEX = ".*\\s+.*";
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validateUser(user);
+        return storageDao.add(user);
+    }
+
+    private void validateUser(User user) {
+        if (user == null) {
+            throw new RuntimeException("User can't be null!");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RuntimeException("User with this login: " + user.getLogin()
+                    + " already exists!");
+        }
+        if (user.getLogin() == null) {
+            throw new RuntimeException("Login can't be null!");
+        }
+        if (user.getLogin().matches(EMPTY_SPACE_REGEX) || user.getLogin().isEmpty()) {
+            throw new RuntimeException("Login can't contain empty space or be empty!");
+        }
+        if (!Character.isLetter(user.getLogin().charAt(0))) {
+            throw new RuntimeException("Login need to begin with letter: " + user.getLogin() + "!");
+        }
```

Comment:
I think it is redundant

Reply:
this one is working, if we try to put first character not letter , than we get exception